### ISSUE TITLE
corrupted double-linked list

### ISSUE THREAD

**Issue by jonsneyers** [Netflix/vmaf#1103]
[inputs.zip](https://github.com/Netflix/vmaf/files/9745201/inputs.zip)

With these inputs, I get the following output:

```
$ ./vmaf -r a.y4m -d b.y4m   --model version=vmaf_v0.6.1neg  --csv -o test.csv --feature float_ssim --feature psnr --feature psnr_hvs --feature float_ms_ssim --feature ciede
VMAF version f266167
1 frame  ⢀⠀ 0.00 FPS
vmaf_v0.6.1neg: 96.433860
corrupted double-linked list
Aborted (core dumped)

$ cat test.csv
Frame,integer_adm2_egl_1,integer_adm_scale0_egl_1,integer_adm_scale1_egl_1,integer_adm_scale2_egl_1,integer_adm_scale3_egl_1,integer_motion2,integer_motion,integer_vif_scale0_egl_1,integer_vif_scale1_egl_1,integer_vif_scale2_egl_1,integer_vif_scale3_egl_1,float_ssim,psnr_y,psnr_cb,psnr_cr,psnr_hvs_y,psnr_hvs_cb,psnr_hvs_cr,psnr_hvs,float_ms_ssim,ciede2000,vmaf,
0,0.996147,0.993235,0.994359,0.996453,0.997946,0.000000,0.000000,0.968577,0.998453,0.998910,0.998964,0.999906,52.481257,52.221434,52.192666,59.845742,54.197084,53.104544,57.698993,0.999873,50.216893,96.433860,
```

This seems to always happen when the reference image is one of these:
https://jon-cld.s3.amazonaws.com/test_images/reference/004.png
https://jon-cld.s3.amazonaws.com/test_images/reference/019.png

For the other reference images I tried, this error does not trigger.

**Comment by kylophone** [comment]
Can't reproduce yet. Can you share how you went from png to y4m, maybe also an md5?

**Comment by kylophone** [comment]
Platform information might be useful too.

**Comment by jonsneyers** [comment]
I went from png to y4m using `ffmpeg -y -loglevel quiet -i intput.png -pix_fmt yuv444p output.y4m`.

Platform is linux (Ubuntu 20.04) on x86_64.

I'm using the linux binary straight from https://github.com/Netflix/vmaf/releases/download/v2.3.1/vmaf

MD5 (a.y4m) = 33cf7e59259da4f73728cec785eaff7e
MD5 (b.y4m) = 1c53cc55331c89bbcef8a5fb99141970
These files are in the zipfile `inputs.zip` I attached above.

To me it happens consistently with that command line. It depends on which `--feature` flags are given though — with default settings (only vmaf) it doesn't crash.
Note that it crashes after writing all the scores to the csv file.
